src/pkg/utils/exec/exec.go

```diff
@@ -165,39 +165,39 @@ func GetOSShell(shellPref types.ZarfComponentActionShell) (string, string) {
 			shell = shellPref.Windows
 		}
 
-		shellArgs = "-Command"
+		shellArgs = append(shellArgs, "-Command", "$ErrorActionPreference = 'Stop';")
 		if shell == "cmd" {
 			// Change shellArgs to /c if cmd is chosen
-			shellArgs = "/c"
+			shellArgs = append(shellArgs, "/c")
 		} else if !IsPowershell(shell) {
 			// Change shellArgs to -c if a real shell is chosen
-			shellArgs = "-c"
+			shellArgs = append(shellArgs, "-e", "-c")
 		}
 	case "darwin":
 		shell = "sh"
 		if shellPref.Darwin != "" {
 			shell = shellPref.Darwin
 		}
 
-		shellArgs = "-c"
+		shellArgs = append(shellArgs, "-e", "-c")
```

Comment:
We should document `set -e` as the default behavior in an admonition in the actions documentation under the `cmd` section (https://github.com/defenseunicorns/zarf/blob/356a4da121ae8cd0ffa720ab17ab859c5a2ac07a/docs/3-create-a-zarf-package/7-component-actions.md?plain=1#L54)